From 8cb35ac7a868a433148e307f2cf19e4b89c39a17 Mon Sep 17 00:00:00 2001
From: Govindraj Raja <Govindraj.Raja@imgtec.com>
Date: Mon, 21 Sep 2015 11:36:42 +0100
Subject: [PATCH 67/70] serial: 8250_dw: Allow clock reconfiguration for baud
 change.

Usage of standard termios function for baud change can change the
UART baud, but if soc's needs clock rate change for associated baud
clock then it is not taken care with default termios function.

However the 8250_serial has a wrapper function which takes care
of this issue, so use the wrapper termios function to set the
clock rate before configuring controller for baud change.

Signed-off-by: Govindraj Raja <Govindraj.Raja@imgtec.com>
---
 drivers/tty/serial/8250/8250_dw.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/tty/serial/8250/8250_dw.c b/drivers/tty/serial/8250/8250_dw.c
index 8435c3f..4264f49 100644
--- a/drivers/tty/serial/8250/8250_dw.c
+++ b/drivers/tty/serial/8250/8250_dw.c
@@ -415,6 +415,7 @@ static int dw8250_probe(struct platform_device *pdev)
 		data->msr_mask_off |= UART_MSR_TERI;
 	}
 
+	p->set_termios = dw8250_set_termios;
 	/* Always ask for fixed clock rate from a property. */
 	device_property_read_u32(p->dev, "clock-frequency", &p->uartclk);
 
-- 
2.7.4

